src/protocol.rs

```diff
@@ -313,42 +324,60 @@ impl std::fmt::Display for ParticipantIdentifier {
+/// messages and incorporate context into proofs. The codebase defines a single
+/// `Identifier` type that is expected to be chosen uniformly at random; this is
+/// used to tag messages. The parameters, participants, and outputs (as well as
+/// the `Identifier`) are incorporated into proofs using a different mechanism
+/// to define the proof context.
 ///
-/// Discrepancy (A): The paper distinguishes between the Session identifiers
-/// (sid) which are created from shared parameters in the protocol at the
-/// beginning of key generation and the Sub-Session identifiers (ssid) that are
-/// created from sid and other post key generation shared parameters for the
-/// purpose of pre/signing. The codebase does not make this kind of distinction
-/// relying instead on a single type for all session identifiers. The
-/// distinction between sessions and sub-sessions is inherently enforced by the
-/// order of inputs and outputs to different stages in the protocol.
+/// 2. The paper distinguishes between sessions (keygen) and sub-sessions
+/// (auxinfo and presign).
+/// The codebase requires the calling application to select a new, unique
+/// session [`Identifier`]s at three points:
+/// (1) immediately before starting a new keygen session;
+/// (2) immediately before starting a new auxinfo session;
+/// (3) immediately before starting a new presigning session.
 ///
-/// Discrepancy (B): The codebase  instantiates [`Identifier`]s in three
-/// different ways: (1) as a session identifier for keygen, that are created at
-/// the start of a key generation instance; (2) as a session identifier for
-/// pre-signing, (3) as a session identifier for auxiliary information
-/// generation sessions; The paper itself only distinguishes between sessions
-/// and sub-sessions and combine (2) and (3).
+///     In fact, we think this is a minor bug in Figure 6 of the paper, since
+/// the definition of `ssid` includes outputs of auxinfo, and thus cannot be
+/// passed as input to auxinfo. We believe the correct instantiation of the
+/// `ssid` for auxinfo is in Figure 3, which includes fixed parameters and
+/// outputs from keygen, but _not_ outputs from auxinfo.
 ///
-/// Discrepancy (C): In the paper ssid is periodically refreshed with the
-/// session key and auxiliary information. The codebase does not do that and
-/// instead relies on the calling application to generate and refresh these
-/// identifiers by randomly sampling a new and unique 32 bytes identifier using
-/// `Identifier::random()`. This assumes that the participants initiating a
-/// protocol run are honestly generating globally unique identifiers and
-/// distributing them to the correct set of parties.
+/// 3. 🔒 In the paper, `ssid` is updated each time the participants run the
+/// key-refresh subprotocol.
+/// The codebase relies on the calling application to generate a new, unique
+/// `Identifier` for each new session.
```

Comment:
Is it worth adding a note that clarifies that the codebase expects the calling application to persist the refreshed key and party-specific parameters, independently from these session identifiers?

Reply:
No, I think that will be distracting here. When we actually implement key refresh, we'll write documentation on that protocol to describe the new storage requirements (e.g. throw out the old, persist the new).